cart/src/App.js

```diff
@@ -0,0 +1,79 @@
+import React, { useEffect, useState } from "react";
+import Cart from "./components/Cart";
+import Home from "./components/Home";
+import NavBar from "./components/Navbar";
+
+const App = () => {
+  const [showScreen, setShowScreen] = useState(true);
+  const [cartArray, setCartArray] = useState([]);
+  const [totalPrice, setTotalPrice] = useState(0);
+
+  const addItemToCart = (items) => {
+   if (!isItemExits(items)) {
+      setCartArray([...cartArray, items]);
+    }
+  };
+
+  const isItemExits = (items) => {
+    const isPresent = cartArray.find((item) => item.id === items.id);
+
+    if (isPresent) {
+      alert("Item is already added to cart");
+    } else {
+      alert("Added to cart");
+    }
+
+    return isPresent;
+  };
+
+  const handleIncDec = (operator, items) => {
+    if (items.quantity > 1 || operator !== -1) {
+      const updatedCart = [...cartArray];
+      const clickedCartIndex = updatedCart.findIndex(
+        (cart) => cart.id === items.id
+      );
+
+      updatedCart[clickedCartIndex].quantity += operator;
+      setCartArray(updatedCart);
+    }
+  };
+
+  const removeCart = (items) => {
+    let updatedCart = [...cartArray];
+    const clickedCartIndex = updatedCart.findIndex(
+      (element) => element.id === items.id
+    );
+    updatedCart[clickedCartIndex].quantity = 1;
+
+    updatedCart = cartArray.filter((element) => element.id !== items.id);
+    setCartArray(updatedCart);
+  };
+
+  const updateTotatPrice = () => {
+    let sum = 0;
+    cartArray.map((element) => (sum += element.prices * element.quantity));
```

Comment:
Can you try using `reduce` instead of `map`. 
`map` is for iterating over an array and returning a transformed version of it
`reduce` is for iterating over an array and reducing it to return the needed piece of info, in our case the `sum` total